libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/NavigationRouteEx.kt

```diff
@@ -113,15 +113,20 @@ fun NavigationRoute.refreshRoute(
                 .build()
         }
     }
+    val newRouteWaypoints = buildNewWaypoints(directionsRoute.waypoints(), waypoints)
+    val newResponseWaypoints = buildNewWaypoints(directionsResponse.waypoints(), waypoints)
     val directionsRouteBlock: DirectionsRoute.() -> DirectionsRoute = {
         toBuilder()
             .legs(updateLegs)
-            .waypoints(buildNewWaypoints(this.waypoints(), waypoints))
-            .updateRouteDurationBasedOnLegsDuration(updateLegs)
+            .waypoints(newRouteWaypoints)
+            .updateRouteDurationBasedOnLegsDurationAndChargeTime(
+                updateLegs = updateLegs,
+                waypoints = newRouteWaypoints ?: newResponseWaypoints
```

Comment:
We kinda assume here that route waypoints have higher priority than response waypoints. While before this logic only lived in `NavigationRoute#waypoints` and IMO it should stay only there. I suggest we do here:
`waypoints = buildNewWaypoints(this@refreshRoute.waypoints, waypoints)`